### PR TITLE
tests: fix typos and misnamed platforms

### LINKED ISSUE #41222: tests: remove not existing platforms from platform allow or integration_platform
**Describe the bug**
I worked on checking of platform existence in this PR: #41081 and in the result I discovered, that among Zephyr tests and samples, there are some not existing platforms in `platform_allow` and `integration_platforms` lists.

I tried to collect all those cases and assign to them person from CODEOWNERS file or from commits history.

People who are mention in following list: please consider if following not existing platforms should be removed from `platform_allow` or `integration_platforms` list or should be replaced by other platform. If some of you feel, that you are not responsible for some of those test please let me know about it - I apologize in advance.

**`platform_allow`**
- [ ] `tests/subsys/storage/stream/stream_flash` | `storage.stream_flash.mpu_allow_flash_write` | `nrf52840_pca10056` | @hakonfam
- [ ] `tests/drivers/build_all/ieee802154` | `ieee802154.build.cc12xx_cc2520` | `cc1352_sensortag` | @JordanYates
- [ ] `tests/kernel/workq/critical` | `kernel.workqueue.nsim` | `nsim_em_mpu_stack_guard` | @nashif @dcpleung @andyross
- [ ] `tests/posix/common` | `portability.posix.common.nsim` `portability.posix.common.newlib.nsim` | `nsim_em_mpu_stack_guard` | @nashif
- [ ] `samples/subsys/ipc/openamp` | `sample.ipc.openamp` | `v2m_musca` | @galak
- [ ] `samples/userspace/syscall_perf` | `sample.syscall_performances` | `riscv32` | @dcpleung @nashif
- [ ] `samples/drivers/counter/maxim_ds3231` | `sample.basic.maxim_ds3231` | `nrf51_pca10028` | @pabigot
- [ ] `samples/boards/arc_secure_services` | `sample.board.arc_secure_services` | `em_starterkit_em7d_secure` | @ioannisg 
- [ ] `samples/tfm_integration/tfm_secure_partition` | `sample.tfm.secure_partition` `sample.tfm.secure_partition.ipc` | `nrf9160dk_nrf9160ns` `v2m_musca_s1_nonsecure` `nrf5340dk_nrf5340_cpuappns` `mps2_an521_nonsecure` | @microbuilder @SebastianBoe 

**`integration_platforms`**
- [ ] `tests/subsys/jwt` | `libraries.encoding.jwt` | `frdm_k64` | @ceolin
- [ ] `tests/drivers/flash` | `drivers.flash.stm32` | `nucleo_f103rc` | @str4t0m 
- [ ] `samples/subsys/ipc/rpmsg_service` | `sample.ipc.rpmsg_service.nrf` | `nrf5340dk_nrf5340_cpuapp bl5340_dvk_cpuapp` | @lairdjm - please convert this into  list of dashes
- [ ] `samples/subsys/ipc/ipc_service/static_vrings_mi` | `sample.ipc.static_vrings_mi.nrf` | `nrf5340dk_nrf5340_cpuapp bl5340_dvk_cpuapp` | @carlocaione - please convert this into  list of dashes

**Expected behavior**
All not existing platforms should be removed or replaced by other platform.

**Environment**
 - OS: Linux
 - Toolchain Zephyr SDK
 - Commit SHA used: 92fbd7da14d1c0c69457441fb13838d0b81f7f12

### PR DESCRIPTION
**Pull request by nashif**
Various obsolote and misnamed platfomrs in test filters theat went
undetected for a while.

Fixes #41222

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

### CODE REVIEW

**Review by erwango — APPROVED** [review]

**Review by microbuilder — APPROVED** [review]

**Review by ceolin — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

File: samples/boards/arc_secure_services/sample.yaml
Comment (gopiotr) [inline samples/boards/arc_secure_services/sample.yaml:9]: Perhaps `em_starterkit_em7d_secure` should be changed just into `em_starterkit_em7d`?

File: tests/subsys/storage/stream/stream_flash/testcase.yaml
Comment (gopiotr) [inline tests/subsys/storage/stream/stream_flash/testcase.yaml:15]: I built and tested this test scenario on real hardware and it failed. 
@hakonfam could you look at this test, before merge this change into main?

**Review by gopiotr — CHANGES_REQUESTED** [review]

File: samples/boards/arc_secure_services/sample.yaml
Comment (nashif) [reply samples/boards/arc_secure_services/sample.yaml:9]: @ruuddw  @abrodkin ^^

Comment (abrodkin) [reply samples/boards/arc_secure_services/sample.yaml:9]: @nashif thanks for pointing to this one. Indeed, back in the day (https://github.com/zephyrproject-rtos/zephyr/pull/9341/) we had that another platform but IIRC we never upstreamed due to missing way of building multi-image application.

Still I'l like to ask @YuguoWH to comment here as well since he's actively working on a similar functionality these days.

**Review by dcpleung — APPROVED** [review]

File: samples/boards/arc_secure_services/sample.yaml
Comment (YuguoWH) [reply samples/boards/arc_secure_services/sample.yaml:9]: There's no `em_starterkit_em7d_secure` now, and we should change it into `em_starterkit_em7d`.

File: tests/subsys/storage/stream/stream_flash/testcase.yaml
Comment (hakonfam) [reply tests/subsys/storage/stream/stream_flash/testcase.yaml:15]: Will do

Comment (hakonfam) [reply tests/subsys/storage/stream/stream_flash/testcase.yaml:15]: This test passes locally on my HW (v2.0.1)

Comment (PerMac) [reply tests/subsys/storage/stream/stream_flash/testcase.yaml:15]: It also works on my local setup.

**Review by PerMac — APPROVED** [review]

File: samples/boards/arc_secure_services/sample.yaml
Comment (ruuddw) [reply samples/boards/arc_secure_services/sample.yaml:9]: agreed, em_starterkit_em7d also supports the SecureShield secure mode.

**Review by ruuddw — CHANGES_REQUESTED** [review]
Thanks, can you please change "em_starterkit_em7d_secure" into "em_starterkit_em7d" instead of removing?

File: tests/subsys/storage/stream/stream_flash/testcase.yaml
Comment (gopiotr) [reply tests/subsys/storage/stream/stream_flash/testcase.yaml:15]: Hmmmm, I was also checking on board v2.0.1 with Zephyr SDK: `zephyr-sdk-0.13.1` and it still failed. I will check my nrfjprog version and other things, but if you wrote, that it works on yours boards I will no longer block this PR. 
@hakonfam @PerMac - thanks for checking :)

**Review by SebastianBoe — APPROVED** [review]

**Review by nvlsianpu — APPROVED** [review]

**Review by gopiotr — APPROVED** [review]

**Review by ruuddw — APPROVED** [review]

**Pull request merged**
